src/Temporalio.Extensions.Hosting/TemporalWorkerClientUpdater.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using Temporalio.Worker;
+
+namespace Temporalio.Extensions.Hosting
+{
+    /// <summary>
+    /// Notification hub that can be used to push Temporal worker client updates to subscribing Temporal workers.
+    /// </summary>
+    public class TemporalWorkerClientUpdater
+    {
+        private readonly object clientLock = new();
+
+        /// <summary>
+        /// The <see cref="EventHandler"/> used to dispatch notifications to subscribers that the Temporal worker client was updated.
+        /// </summary>
+        public event EventHandler<TemporalWorkerClientUpdatedEventArgs> TemporalWorkerClientUpdated
```

Comment:
I think this can just be an `internal` property and therefore doesn't need `EventArgs` class (even though that's how we'd do it if this were public API). I am just hoping to keep the public API and general code surface area lower. Since users are the ones invoking update, I don't think we need to expose this. I don't think users need to test that Temporal's client updating mechanism works, at least not enough to make all of this public API.